### PR TITLE
Restricted competitions support

### PR DESCRIPTION
**Pull request by pedrosino**
NOT READY, THIS IS JUST AN EXPERIMENT

TO-DO
* messages at registration page
- [ ] if already registered for another restricted comp, inform that cannot register for this one
- [ ] also offer to cancel the other registration
- [ ] if not registered for another restricted comp, inform that can only participate in one of them

restricted1
restricted2

### CODE REVIEW

File: WcaOnRails/app/controllers/competitions_controller.rb [hunk 1]
```diff
@@ -299,6 +300,7 @@ def time_until_competition
 
   def show
     @competition = competition_from_params
+    @nearby_competitions = get_nearby_competitions(@competition)
```
Comment (jfly) [inline WcaOnRails/app/controllers/competitions_controller.rb]: Probably shouldn't be doing this query unless we're actually going to be using it.

File: WcaOnRails/app/views/competitions/_competition_info.html.erb [hunk 2]
```diff
@@ -82,4 +82,15 @@
     <% end %>
 
   </div>
+  <div class="col-md-8">
+    <% if competition.restricted %>
```
Comment (jfly) [inline WcaOnRails/app/views/competitions/_competition_info.html.erb]: nit: `competition.restricted` -> `competition.restricted?`

**Comment by jfly** [comment]
Hey @pedrosino,

Any update here? Is the US northeast going to be switching to this system soon?

**Comment by pedrosino** [comment]
Sorry, life has been a little crazy recently. I was hoping Tim Reynolds would take a look here, but apparently he couldn't either.

**Comment by jfly** [comment]
(cc-ing @timreyn so he knows about this, and maybe can provide some insight into when this needs to be done)

**Comment by timreyn** [comment]
Hey, thanks for cc-ing me.

I think Sebastien viewed having all competitions using WCA site for registration as a prerequisite for actually having restricted competitions.  Otherwise we would need to communicate between WCA and CUSA and other sites as to who can register for what.  So I think this is blocked on payments being implemented and US comps moving over.

**Comment by jfly** [comment]
> So I think this is blocked on payments being implemented and US comps moving over.

Just to be clear, this PR is not blocked on payments being implemented, that's just something competitions in the US northeast will want before they switch to the WCA website for competition registration.

In the mean time, we need to update our existing PHP code to actually support this, and I bet that's not something anyone on the software team is excited to do. @timreyn, I think you're going to have to either implement this yourself (will gladly accept PRs), or convince someone on the software team to work on that. Maybe @pedrosino is willing to do it, since he did create this PR =)

**Comment by jfly** [comment]
Alternatively, if you're willing to wait for us to port all the relevant PHP scripts to Rails, I will gladly implement this in Ruby myself. However, the PHP -> Rails port has been going on for almost 2 years, with no end in sight =(

**Comment by pedrosino** [comment]
> In the mean time, we need to update our existing PHP code to actually support this

What part of the php code do you mean?

**Comment by jfly** [comment]
Don't we have to update our results posting code to check for violations of the restricted competitions policy? It's nice that we're checking for this at registration time, but the only way to be *sure* is to check at results posting time (for example, walk in competitors who didn't register online).

**Comment by pedrosino** [comment]
Ah, right.

**Comment by jfly** [comment]
I'm going to close this. This is a nice to have, but the thing that actually needs to be done is to fix our results posting scripts.

**Pull request closed without merging**